File: GA360.DAL.Entities/Entities/Customer.cs
using GA360.DAL.Entities.BaseEntities;
using System.ComponentModel.DataAnnotations;
using static GA360.DAL.Entities.Enums.StatusEnum;

namespace GA360.DAL.Entities.Entities;

public class Customer : Audit, IModel
{
    [Required]
    public string FirstName { get; set; }
    [Required]
    public string LastName { get; set; }
    [Required]
    public string Contact { get; set; }
    public string FatherName { get; set; }
    public string About { get; set; }
    public string Gender { get; set; }
    public string Email { get; set; }
    public string Role { get; set; }
    public string? Description { get; set; }
    public string Location { get; set; }
    public Status Status { get; set; }
    public int CountryId { get; set; }
    public virtual Country Country { get; set; }
    public Guid? TenantId { get; set; }
    public int Id { get; set; }
    public int AddressId { get; set; }
    public string DOB {  get; set; }
    public string NI { get; set; }
    public string Disability { get; set; }
    public string EmploymentStatus { get; set; }
    public string Employer { get; set; }
    public string ePortfolio { get; set; }
    public int EthnicOriginId {  get; set; }
    public string? AvatarImage { get; set; }
    public int? TrainingCentreId { get; set; }
    public virtual Address Address { get; set; }
    public virtual EthnicOrigin EthnicOrigin { get; set; }
    public virtual List<CustomerSkills> CustomerSkills { get; set; }
    public virtual List<QualificationCustomerCourseCertificate>? QualificationCustomerCourseCertificates { get; set; }
    public virtual TrainingCentre TrainingCentre { get; set; }

}

File: ga360.client/src/types/customer.ts
import { Gender } from 'config';
import { extend } from 'lodash';
import { CertificateModel, CourseModel, DocumentFileModel, QualificationModel } from './customerApiModel';

export interface CustomerProps {
  modal: boolean;
}

export interface CustomerList {
  id?: number;
  avatar: number;
  firstName: string;
  lastName: string;
  fatherName: string;
  name: string;
  email: string;
  age: number;
  gender: Gender;
  role: string;
  orders: number;
  progress: number;
  status: number;
  orderStatus: string;
  contact: string;
  country: string;
  location: string;
  about: string;
  skills: string[];
  time: string[];
  date: Date | string | number;
}

export interface CustomerListExtended extends CustomerList {
  avatarImage: string | null;
  dateOfBirth: string;
  ethnicity: string;
  disability: string;
  employeeStatus: string;
  employer: string;
  trainingCentre: number;
  nationalInsurance: string;
  portfolio: string;
  dob: string;
  street: string;
  city: string;
  number: string;
  postcode: string;
  documents: string[];
  fileDocuments: DocumentFileModel[];
  courses: CourseModel [];
  qualifications: QualificationModel [];
  certificates: CertificateModel [];
  trainingCentreId?: number; // Make this property optional
  employmentStatus: string;
}

export interface BasicCustomer {
  id: number;
  email: string;
}

export interface User {
  firstName: string;
  lastName: string;
  email: string;
  customerId: number;
  role: string;
  roleId: number;
  trainingCentreId: number;
  contact: string;
  city: string;
  avatarImage: string;
  employeeStatus: string;
}